snuba/query/parser/expressions.py

```diff
@@ -21,22 +22,26 @@
 # This root element is needed because of the ambiguity of the aggregation
 # function field which can mean a clickhouse function expression or the simple
 # name of a clickhouse function.
-root_element    = function_call / function_name
-expression      = function_call / simple_term
-parameters_list = parameter* (expression)
-parameter       = expression space* comma space*
-function_call   = function_name open_paren parameters_list? close_paren (open_paren parameters_list? close_paren)?
-simple_term     = quoted_literal / numeric_literal / column_name
-literal         = ~r"[a-zA-Z0-9_\.:-]+"
-quoted_literal  = "'" string_literal "'"
-string_literal  = ~r"[a-zA-Z0-9_\.:-]*"
-numeric_literal = ~r"-?[0-9]+(\.[0-9]+)?(e[\+\-][0-9]+)?"
-column_name     = ~r"[a-zA-Z_][a-zA-Z0-9_\.]*"
-function_name   = ~r"[a-zA-Z_][a-zA-Z0-9_]*"
-open_paren      = "("
-close_paren     = ")"
-space           = " "
-comma           = ","
+
+root_element          = low_pri_arithmetic
```

Comment:
I think these names are a bit misleading, expecially the difference between expression and the term in an arithmetic operation:
why not:
```
root_element = low_pri_arithmetic
low_pri_arithmetic = ... as it is today
high_pri_arithemtic = space* / arithmetic_term / ....the rest is how it is today
arithmetic_term = function_call / numeric_literal / column_name
...
param_expression = low_pri_arithmetic / quoted_literal
parameters_list       = parameter* (param_expression)
parameter             = param_expression space* comma space*
....
```

Reply:
Done.